build/azuredevops/azure-pipelines-release.yml

```diff
@@ -0,0 +1,144 @@
+    pool:
+      name: AzurePipelines-EO
+      demands:
+        - ImageOverride -equals 1ESPT-Ubuntu22.04
+      os: Linux
+    customBuildTags:
+      - ES365AIMigrationTooling
+    stages:
+      - stage: BuildVsix
+        jobs:
+          - job: build
+            displayName: Build VSIX
+            timeoutInMinutes: 720
+            templateContext:
+              outputs:
+                - output: pipelineArtifact
+                  displayName: 'publish vsix artifact'
+                  targetPath: build_output
+                  artifactName: $(ARTIFACT_NAME_VSIX)
+            steps:
+              - checkout: self
+                clean: true
+                submodules: true
+                fetchTags: true
+                persistCredentials: True
+              - task: MicroBuildSigningPlugin@4
+                inputs:
+                  signType: '$(SignType)'
+                  feedSource: 'https://devdiv.pkgs.visualstudio.com/DefaultCollection/_packaging/MicroBuildToolset/nuget/v3/index.json'
+                env:
+                  TeamName: '$(TeamName)'
+              - task: NodeTool@0
+                displayName: Use Node 18.x
+                inputs:
+                  versionSpec: 18.x
+              - task: CmdLine@2
+                displayName: npm install
+                inputs:
+                  script: npm run install:all
+              - task: CmdLine@2
+                displayName: Package VSIX
+                inputs:
+                  script: |
+                    npm run package
+                    mv pyright-*.vsix $(VSIX_NAME)
+                  workingDirectory: packages/vscode-pyright
+              - task: CopyFiles@2
+                displayName: 'Copy vsix to: build_output'
+                inputs:
+                  SourceFolder: packages/vscode-pyright
+                  Contents: '*.vsix'
+                  TargetFolder: build_output
+      - stage: CreateRelease
+        dependsOn:
+          - BuildVsix
+        jobs:
+          - job: create_release
+            displayName: Create GitHub Release
+            steps:
+              - checkout: none
+              - download: current
+              - task: GitHubRelease@1 #https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/github-release-v1?view=azure-pipelines
+                displayName: 'Create GitHub Release'
+                inputs:
+                  gitHubConnection: 'Github-Pylance' # The name of the GitHub service connection
+                  repositoryName: 'microsoft/pyright' # The name of your GitHub repository
+                  action: 'create'
+                  isDraft: true
+                  isPreRelease: false
+                  addChangeLog: true
+                  title: 'Published $(Build.SourceBranchName)'
+                  assets: |
+                    $(Pipeline.Workspace)/$(ARTIFACT_NAME_VSIX)/$(VSIX_NAME)
+      - stage: WaitForValidation

```

Comment:
Will Eric have access to the AzDO page to tell the release to proceed? I'm not sure if he currently does any manual testing of the VSIX before publishing to Marketplace. He may prefer to skip the manual validation step and go straight to publishing.

Reply:
No he wont have access.. it will be up to us for final button click

Reply:
Ok, is that an intentional process change for security reasons?

Reply:
not intentional... i just dont think he has azure devops access. Maybe there is a way to grant it

Reply:
Ok. So I'm wondering if removing the manual validation step would be simpler and closer to what he's used to.

Reply:
i think i need to leave it though.  it can be changed to 'pass' when the timeout expires... Eric needs time to change the draft release to publish and edit the notes.  In azure devops there is no build trigger or github published.

Reply:
I see. Could the GitHub release be created as a non-draft? For what it's worth, the GitHub release text can be changed after the release is published.

On the other hand, maybe publishing to marketplace just because a tag was pushed is too risky and a manual step somewhere along the line would be best.